File: Easy_problems/isSubsequence/Sol.py
class Solution(object):
    def isSubsequence(self, s, t):
        j = 0
        
        if s == "":
            return True
        
        
        for i in range(0, len(t)):
            if t[i] == s[j]:
                j = j+1
                if j == len(s):
                    break
                
        return len(s) == j

File: Easy_problems/isSubsequence/Sol.java
public class Sol{
    public static void main(String[] args) {
        System.err.println(isSubsequence("abc", "aeblc"));
    }
    public static boolean isSubsequence(String s, String t) {
        char [] sub = s.toCharArray(), str = t.toCharArray();
        if(sub.length==0)
            return true;
            
        int j = 0;
        for(int i=0; i<str.length;i++)
            if(sub[j] == str[i]){
                j++;
                if (j == sub.length)
                    break;
            }


        return j == sub.length;
    }
}